cli/internal/commands/check/experiment.go

```diff
@@ -283,12 +269,36 @@ func checkBaseline(lint logr.Logger, p *optimizev1beta2.Parameter) {
 	}
 }
 
-func metricQueryDryRun(m *optimizev1beta2.Metric) (string, string, error) {
+func checkQuery(lint logr.Logger, m *optimizev1beta2.Metric) {
+	switch m.Type {
```

Comment:
Is there a reason we wouldn't consolidate this into a single switch statement?

Reply:
They are looking at different things at different times: the first switch looks at the raw query prior to evaluation (and we may return from it); the second switch looks at the evaluated query (template evaluated against an empty object).